bin/fin

```diff
@@ -1747,24 +1758,29 @@ docker_machine_mount_smb ()
 	# use his password for mount -t cifs ...
 
 	# Get a list of logical drives (type 3 = Local disk).
-	local DRIVES=$(wmic logicaldisk get drivetype,name | grep 3 | awk '{print $2}' | sed 's/\r//g')
-	read -s -p "Enter your Windows account password: " PASSWORD
+	local drives=$(wmic logicaldisk get drivetype,name | grep 3 | awk '{print $2}' | sed 's/\r//g')
+	read -s -p "Enter your Windows account password: " password
 	echo # Add a new line after user input.
 
-	for DRIVE in ${DRIVES}; do
-		local MOUNT_POINT=$(cygpath -u "$DRIVE" | sed 's/^\/cygdrive\///')
-		smb_share_add "$MOUNT_POINT" "$DRIVE" && \
-			smb_share_mount "$MOUNT_POINT" "/$MOUNT_POINT" "$PASSWORD" || \
+	for drive in ${drives}; do
+		local mount_point=$(cygpath -u "$drive" | sed 's/^\/cygdrive\///')
+		# Avoid conflicts with existing drive shares buy using a unique share name
+		local share_name="docksal-$mount_point"
+		smb_share_add "$share_name" "$drive" && \
+			smb_share_mount "$share_name" "/$mount_point" "$password" || \
 			(echo-red "Error creating share" && exit 1)
 	done
 }
 
 docker_machine_remove_smb ()
 {
-	local DRIVES=$(wmic logicaldisk get drivetype,name | grep 3 | awk '{print $2}' | sed 's/\r//g')
-	for DRIVE in ${DRIVES}; do
-		local MOUNT_POINT=$(cygpath -u "$DRIVE" | sed 's/^\/cygdrive\///')
-		command_share="net share ${MOUNT_POINT} /DELETE"
+	local drives=$(wmic logicaldisk get drivetype,name | grep 3 | awk '{print $2}' | sed 's/\r//g')
+	for drive in ${drives}; do
+		local mount_point=$(cygpath -u "$drive" | sed 's/^\/cygdrive\///')
+		# Avoid conflicts with existing drive shares buy using a unique share name
+		local share_name="docksal-$mount_point"
+		# TODO: remove legacy style shares removal (net share ${mount_point} /DELETE) in September 2017
+		command_share="net share ${mount_point} /DELETE & net share ${share_name} /DELETE"
```

Comment:
I'm not 100% sure whether to add this or not.

There is a chance we will be dropping valid shares users want to have.
On the other side, if not doing this, we will be leaving around those legacy shares.